deploy-deps.sh

```diff
@@ -44,6 +46,13 @@ deploy() {
     deploy_registry
 }
 
+test_pvc_binding(){
+    local pvc_resources="${script_path}/dependencies/pre-deployment-pvc-binding"
+    echo "Creating PVC from '$pvc_resources' using the cluster's default storage class"
+    kubectl apply -k "$pvc_resources"
+    set -e; kubectl wait --for=jsonpath='{status.phase}'=Bound pvc/test-pvc -n test-pvc-ns --timeout=20s
```

Comment:
We should not have the `set -e` here and we should use `retry`.
```suggestion
    retry "kubectl wait --for=jsonpath='{status.phase}'=Bound pvc/test-pvc -n test-pvc-ns --timeout=20s"
```

Reply:
Why not fail the script if we can't bind PVC? We wanted it to `exit 1` before using retry

Reply:
We have multiple such statements which are failing the script without setting this option just before running them. Is this statement any different? What's the reason for it to need this while others don't?

Reply:
Simply the fact it does not fail the script:
```
error: timed out waiting for the condition on persistentvolumeclaims/test-pvc
namespace "test-pvc-ns" deleted
persistentvolumeclaim "test-pvc" deleted
pod "test-pvc-consumer" deleted
Deploying Konflux Dependencies
namespace/cert-manager unchanged
customresourcedefinition.apiextensions.k8s.io/certificaterequests.cert-manager.io unchanged
customresourcedefinition.apiextensions.k8s.io/certificates.cert-manager.io unchanged
....
```

Reply:
AFAIK, The other statements shouldn't fail the script unless we explicitly act upon their return status

Reply:
I think we should fail the script in case a PVC cannot be created.
There is no point in moving on further.

Reply:
Right. I'm not sure what caused it to continue, but I removed the pod for the sake of testing it and it indeed failed after the timeout reached.  
fixed